### PR TITLE
feat(index): add context to `outputPath` && `publicPath`

### PR DESCRIPTION
**Pull request by lsycxyj**
I propose to add issuerContext to outputPath and publicPath to allow users change the resolved paths dynamically (eg. implementing relative path by myself).
BTW,  the outputPath should be able to be changed by the users after using "useRelativePath" because it will emit files to wrong paths (eg. files will be placed to parent folder of output path when it starts with "..").

### CODE REVIEW

**Comment by lsycxyj** [comment]
Knock, knock. Nobody's here?

**Comment by adriancmiranda** [comment]
@lsycxyj This will being resolved in #150 😉

**Comment by lsycxyj** [comment]
@adriancmiranda Thank you. I didn't mean that I only wanna fix the bug, but also wanna let the users handle the paths by themselves.

**Comment by michael-ciniawsky** [comment]
@lsycxyj Sry we are shaping up a few things behind the scences in terms of CSS handling and this likely affects `file/url-loader` aswell, so they get stalled a bit atm, please be patient 😛

**Comment by lsycxyj** [comment]
@michael-ciniawsky I'm looking forward to your good news. : )

**Comment by michael-ciniawsky** [comment]
@lsycxyj Can you please provide a small example for this, just to clarify ? :)

File: index.js [hunk 1]
```diff
@@ -38,8 +38,8 @@ module.exports = function(content) {
 	var outputPath = "";
 
 	var filePath = this.resourcePath;
+	var issuerContext = this._module && this._module.issuer && this._module.issuer.context || context;
```
Comment (michael-ciniawsky) [inline index.js]: nitpick `issuerContext` => `issuer || context` (personally `context`)

Comment (bebraw) [inline index.js]: Yeah, better clean this up.

Comment (lsycxyj) [inline index.js]: @michael-ciniawsky Well... Actually this line of code is from the `master` branch. I just moved it outside the block. Do you mean I should choose `issuer` or `context` to pass to the `publicPath` function?

Comment (lsycxyj) [inline index.js]: @michael-ciniawsky My purpose of adding issuer's context to the `publicPath` function is to give the   user a way to the resolved path trickily (eg. I can give a new relative path according to the issuer's context). So I prefer the issuer's context personally.

But I didn't fix the "useRelativePath" bug in this patch.

I notice that all the test cases are mocked cases, and there's no (good) way to find out problems caused by the webpack APIs (eg. `this._module`).
I'm not familiar with the webpack APIs. Maybe I need some help or advice to test it in a better way.

BTW, `this._module` has been deprecated by [officual docs](https://doc.webpack-china.org/api/loaders/#this-_module). Maybe it should be changed in some other way?

**Comment by d3viant0ne** [comment]
@lsycxyj - I have set this to blocked ( #165 ) & moved this feature into `1.1.0`, the next minor range after the pending `1.0.0` release.

**Comment by evilebottnawi** [comment]
@lsycxyj friendly ping

**Comment by lsycxyj** [comment]
@evilebottnawi Is there anything I need to do?

**Comment by evilebottnawi** [comment]
Done in https://github.com/webpack-contrib/file-loader/pull/305

**Comment by lsycxyj** [comment]
My idea was merged into the library but I'm not in the contributors list. :cry:

**Comment by evilebottnawi** [comment]
@lsycxyj what do you mean?

**Pull request closed without merging**